src/commands/installDocker.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IActionContext } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import LocalOSProvider from '../utils/LocalOSProvider';
+import { DockerInstallerBase } from './dockerInstallerBase';
+
+class WindowsDockerInstaller extends DockerInstallerBase {
+    protected downloadUrl: string = 'https://aka.ms/download-docker-windows-vscode';
+    protected fileExtension: string = 'exe';
+    protected installationMessage: string = localize('vscode-docker.commands.WindowsDockerInstaller.installationMessage', 'Installer is launched. Please follow the prompts to complete the installation, and then start the Docker Desktop.');
+    protected getInstallCommand(fileName: string): string {
+        // Windows require double quote.
+        return `"${fileName}"`;
+    }
+}
+
+class MacDockerInstaller extends DockerInstallerBase {
+    protected downloadUrl: string = 'https://aka.ms/download-docker-mac-vscode';
+    protected fileExtension: string = 'dmg';
+    protected installationMessage: string = localize('vscode-docker.commands.MacDockerInstaller.installationMessage', 'Installer is launched. Please follow the prompts to complete the installation, and then start the Docker Desktop.');
+    protected getInstallCommand(fileName: string): string {
+        return `chmod +x '${fileName}' && open '${fileName}'`;
```

Comment:
Is that required in Mac (the chmod part)? I tried with just the "open" command and it works fine.

Reply:
I added it based on what Devin ran into. He got access denied or something like that. I think it is safe to add execute permission.